### ISSUE TITLE
empty poco var

### ISSUE THREAD

**Issue by quenenni** [husisusi/officeonlin-install.sh#110]
2 times I started the script on a fresh debian 8 and 2 times the script stopped when starting the poco stuff:
```
--2017-08-09 05:22:00--  https://pocoproject.org/releases/poco-/poco-1.7.8p3-all.tar.gz
Résolution de pocoproject.org (pocoproject.org)… 162.209.7.4
Connexion à pocoproject.org (pocoproject.org)|162.209.7.4|:443… connecté.
requête HTTP transmise, en attente de la réponse… 404 Not Found
2017-08-09 05:22:01 erreur 404 : Not Found.
```

The '${poco_version_folder}' var seems to be empty when composing the url to wget the tar file.

**Comment by quenenni** [comment]
What is strange is that when I modified the script to echo the var $poco_version_folder (line 105) and then exit the script, the echo shows the correct value (1.7.8).
And after that line, that var is not modified / used up to the line where it's empty.. Very strange.

**Comment by aalaesar** [comment]
could be a context problem created by the logs system using unamed function and output redirection
😕 
but the variable should be global for all children processes

**Comment by aalaesar** [comment]
maybe you can try to echo the variable at different place in the script to see when it get unsetted.
PS: are you using Debian ?

**Comment by quenenni** [comment]
I just tested the script by bypassing the libreoffice compilation section.. The var $poco_version_folder was good and the poco stuff started.

**Comment by quenenni** [comment]
Yes, Debian 8.9, downloaded yesterday.

I'll try that on my next attemps.

**Comment by aalaesar** [comment]
I wonder if the new script's release would have the same issue in debian.
I haven't yet encounter this issue in debian tests.
Will you test once the merge request is done ?

**Comment by quenenni** [comment]
Ok.. third time was the charm..
Everything went smoothly.

The difference is that I downloaded the last version of the script for this run. Maybe the changes from yesterday did something, but not sure.

What was really strange is that the others variables, set at the same place as the other didn't loose their value.. Why just one..

Anyway, the problem is closed for me. I have my first lool without docker, without CODE and without the limitations and I'm very happy.
I'll try now on a Debian 9

Thanks for your help.

**Comment by quenenni** [comment]
@aalaesar debian tests? what debian tests?
Also which merge request? The one that separates the script in lot of smaller files?

**Comment by aalaesar** [comment]
@quenenni 
indeed I was talking about #111 
I also test each release on debian & ubuntu hosts but i'm stuck with debian 8 for now.

**Comment by quenenni** [comment]
Unfortunately, this problem occurred also with the new script on a fresh debian 9 installed.
Then I launched again the script and it worked.

So it seems to happen when you have the full compilation.

As I have to restart the all process with my debian 9, I'll put echos in the file for the first compilation.

**Comment by quenenni** [comment]
@aalaesar My last tests (the last one with Debian 8 and the 2 last ones with Debian 9) was done with the version that include #111 

For the testing stuff, how can I help you? 
Are you talking about running a number of scripts on the server? 
Or accessing test files (like I'm talking here #114 )?

**Comment by aalaesar** [comment]
Hi,
Following your observations I think something in the Lo compilation process make debian's bash either change context or unset the variable. 
This need further investigation.
You can add an echo of the poco_* variables in strategic points  (lines 126 - 131 - 133 - 136 ) of the main script. Something like
`echo "debug issue 110, poco: $poco"` so you can find them easier in the logs.
You can use option ` -f lo` to force compilation for libre office core.
Regards

**Comment by quenenni** [comment]
It's already on it's way :)

Unfortunately, it already seems to have a problem.
Up to this point, all my "echo $poco_version_folder >> /tmp/debug" show an empty value?! 
That's even before the compilation process starts.
```
######################## libreoffice compilation ##############################
echo "0: "$poco_version_folder > /tmp/debug
{ 
echo "1: "$poco_version_folder >> /tmp/debug 
# shellcheck source=/project/bin/corePrep.sh
source "$ScriptFullPath/bin/corePrep.sh"
echo "2: "$poco_version_folder >> /tmp/debug
} > >(tee -a ${log_dir}/LO-compilation.log) 2> >(tee -a ${log_dir}/LO-compilation.log >&2)
echo "apres: "$poco_version_folder >> /tmp/debug
# shellcheck source=/project/bin/coreBuild.sh
echo "avant coreBuild: "$poco_version_folder >> /tmp/debug
source "$ScriptFullPath/bin/coreBuild.sh"
echo "apres coreBuild "$poco_version_folder >> /tmp/debug
```
```
0:
1:
2:
apres:
avant coreBuild:
```
And I executed again several time the script (I added an exit statement before the compilation section) and each time now, I have the correct value of the var.
```
0:1.7.8
1:1.7.8
...
```

That's just silly :)

Did you change something in the script for this problem?

**Comment by aalaesar** [comment]
Could it be that the host fails silently to fetch the poco website and therefore the value are empty  ?
what's  your  test process ?
Do you run automation  script  to create the host from an image and procede  to run the script  ?
Just trying  to understand here, not making accusations 😉.
On a brand new host,
- you can try to set the value  out from the script  so you can see if the whole pipes curl command work 
- You can try to add the echo  just after loading the config files in the script  to check  the value is present.

**Comment by quenenni** [comment]
I'm using virtual Box.
I install a debian 9 netinstall (downloaded 4 days ago)
I select "openssh server" and "System utilities" in tasksel.
When the installation is finished, I do:
```
apt install vim htop net-tools screen unzip
wget https://github.com/husisusi/officeonlin-install.sh/archive/master.zip
unzip master.zip
rm master.zip
vim officeonlin-install.sh-master/bin/systemSetup.sh
  -> there, I do the small changes I need for Debian 9 (see the ticket related to it, but it's just removing packages from the install list and changing the jdk version)
vim /etc/apt/sources.lst
  -> add the 2 lines for jessie-backports
vim /etc/apt/preferences.d/00_ssl
  -> add the preference to jessie-backports for ssl packages

officeonlin-install.sh-master/bin/systemSetup.sh (starting the script)
```
That's all I did this time.

In this attempt, instead of installing libpng12 from wheezy, I changed libpng12.0 by libpng16.16 available in Stretch (it's the new name I think).. 
I'll see if my idea was good when the compilation will be finished.

I didn't have any network problem (my first command is an apt install)

Ps: ask what you need, I won't feel accused :P
I want to understand this nonsense too..

**Comment by quenenni** [comment]
Hooo, you know what..

Curl is installed by the script.
That means the first time I ran the script, as my server has nearly nothing installed, curl included, the curl command return an error (command not found).

And all the times after, curl is installed when starting the script.

**Comment by Kassiematis** [comment]
defaults.cfg is accessed before systemsetup.sh. should be the other way around.

**Comment by quenenni** [comment]
I think this error is different.
When you see my first post here, you can see that the value of the var poco_version_folder was empty, but the value of poco_version was good.
And poco_version is taken from poco_version_latest that is itself using curl to get the value.

So, on that test, curl was already installed.

It was on my Debian 8.
In Debian 8, when you select "Ssh server" and "System utilities" when installing, you have something like 250 packages.
On Debian 9, it's only 128 packages.

It seems that they decided to remove lots of packages from the basic install.
I was already surprised to discover that the command ifconfig is not available without installing the net-tools package on 9.
So lots of commands that were available directly aren't anymore (like curl maybe, not sure tough).

Anyway, I started a new compilation process on a freshly installed Debian 9.
I installed curl before running the script and now my echos are returning the correct value (up to the git clone section where it is atm)

I'll have to wait +-6 hours for the process to finish now and see what's what.

**Comment by quenenni** [comment]
Well, I'm still very confused about this..

On my last attempt, from a newly installed Debian 9 with Curl installed before launching the script, everything worked perfectly.
No problem with the var poco_version_folder

You just have to fix the Curl problem, but for the rest, let's forget it momentarily because I have no idea what happened before.

**Comment by aalaesar** [comment]
well done @quenenni  👍 
if you still confused about why the poco_version_folder variable was unset, in fact I remembered I did an ad-hoc fix month ago when I encountered the same problem.
@Kassiematis: the setup script use configuration variables. I just can add a curl installation just after the root test as an add-hoc fix
A more aesthetic solution could be adding a bootstrap script that install curl and create the list of packages to be installed by the setup script, depending on the distribution and version  detected.
